From 3c3c02cdd0298cd27b4d7bf0a59b006756823cc3 Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Wed, 7 Nov 2018 12:36:37 +0200
Subject: [PATCH 5056/5242] MLK-20181-8: drm/mxsfb: Cleanup after upstream
 backport

commit  c8e6ebf1c6577db32de5e07bab89e37f63dc6d34 from
https://source.codeaurora.org/external/imx/linux-imx.git

Patches related to suspend/resume have been backported from upstream
kernel, therefore some of the mxsfb_drm_private members are no longer
needed, so remove them, among with the code around them.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Acked-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/mxsfb/mxsfb_crtc.c |   42 +++++++-----------------------------
 drivers/gpu/drm/mxsfb/mxsfb_drv.h  |    4 ----
 2 files changed, 8 insertions(+), 38 deletions(-)

diff --git a/drivers/gpu/drm/mxsfb/mxsfb_crtc.c b/drivers/gpu/drm/mxsfb/mxsfb_crtc.c
index 66a8717..ab8e2aa 100644
--- a/drivers/gpu/drm/mxsfb/mxsfb_crtc.c
+++ b/drivers/gpu/drm/mxsfb/mxsfb_crtc.c
@@ -409,22 +409,12 @@ static void mxsfb_crtc_mode_set_nofb(struct mxsfb_drm_private *mxsfb)
 
 	writel(SET_DOTCLK_H_VALID_DATA_CNT(m->hdisplay),
 	       mxsfb->base + LCDC_VDCTRL4);
-
-	if (mxsfb->gem != NULL) {
-		writel(mxsfb->gem->paddr,
-		       mxsfb->base + mxsfb->devdata->next_buf);
-		mxsfb->gem = NULL;
-	}
-
 }
 
 void mxsfb_crtc_enable(struct mxsfb_drm_private *mxsfb)
 {
 	dma_addr_t paddr;
 
-	if (mxsfb->enabled)
-		return;
-
 	if (mxsfb->devdata->flags & MXSFB_FLAG_BUSFREQ)
 		request_bus_freq(BUS_FREQ_HIGH);
 
@@ -440,22 +430,15 @@ void mxsfb_crtc_enable(struct mxsfb_drm_private *mxsfb)
 	}
 
 	mxsfb_enable_controller(mxsfb);
-
-	mxsfb->enabled = true;
 }
 
 void mxsfb_crtc_disable(struct mxsfb_drm_private *mxsfb)
 {
-	if (!mxsfb->enabled)
-		return;
-
 	mxsfb_disable_controller(mxsfb);
 	mxsfb_disable_axi_clk(mxsfb);
 
 	if (mxsfb->devdata->flags & MXSFB_FLAG_BUSFREQ)
 		release_bus_freq(BUS_FREQ_HIGH);
-
-	mxsfb->enabled = false;
 }
 
 void mxsfb_plane_atomic_update(struct mxsfb_drm_private *mxsfb,
@@ -463,11 +446,9 @@ void mxsfb_plane_atomic_update(struct mxsfb_drm_private *mxsfb,
 {
 	struct drm_simple_display_pipe *pipe = &mxsfb->pipe;
 	struct drm_crtc *crtc = &pipe->crtc;
-	struct drm_device *drm = crtc->dev;
 	struct drm_framebuffer *fb = pipe->plane.state->fb;
 	struct drm_framebuffer *old_fb = old_state->fb;
 	struct drm_pending_vblank_event *event;
-	struct drm_gem_cma_object *gem;
 	dma_addr_t paddr;
 
 	spin_lock_irq(&crtc->dev->event_lock);
@@ -483,15 +464,15 @@ void mxsfb_plane_atomic_update(struct mxsfb_drm_private *mxsfb,
 	}
 	spin_unlock_irq(&crtc->dev->event_lock);
 
-	if (!fb)
-		return;
-
-	gem = drm_fb_cma_get_gem_obj(fb, 0);
+	paddr = mxsfb_get_fb_paddr(mxsfb);
+	if (paddr) {
+		mxsfb_enable_axi_clk(mxsfb);
+		writel(paddr, mxsfb->base + mxsfb->devdata->next_buf);
+		mxsfb_disable_axi_clk(mxsfb);
+	}
 
-	if (!mxsfb->enabled) {
-		mxsfb->gem = gem;
+	if (!fb || !old_fb)
 		return;
-	}
 
 	/*
 	 * TODO: Currently, we only support pixel format change, but we need
@@ -500,17 +481,10 @@ void mxsfb_plane_atomic_update(struct mxsfb_drm_private *mxsfb,
 	if (old_fb->format->format != fb->format->format) {
 		struct drm_format_name_buf old_fmt_buf;
 		struct drm_format_name_buf new_fmt_buf;
-		DRM_DEV_DEBUG_DRIVER(drm->dev,
+		DRM_DEV_DEBUG_DRIVER(mxsfb->dev,
 				"Switching pixel format: %s -> %s\n",
 				drm_get_format_name(old_fb->format->format, &old_fmt_buf),
 				drm_get_format_name(fb->format->format, &new_fmt_buf));
 		mxsfb_set_pixel_fmt(mxsfb, true);
 	}
-
-	paddr = mxsfb_get_fb_paddr(mxsfb);
-	if (paddr) {
-		mxsfb_enable_axi_clk(mxsfb);
-		writel(paddr, mxsfb->base + mxsfb->devdata->next_buf);
-		mxsfb_disable_axi_clk(mxsfb);
-	}
 }
diff --git a/drivers/gpu/drm/mxsfb/mxsfb_drv.h b/drivers/gpu/drm/mxsfb/mxsfb_drv.h
index b4c98d4..6db24dc 100644
--- a/drivers/gpu/drm/mxsfb/mxsfb_drv.h
+++ b/drivers/gpu/drm/mxsfb/mxsfb_drv.h
@@ -43,10 +43,6 @@ struct mxsfb_drm_private {
 	struct drm_panel		*panel;
 	struct drm_bridge		*bridge;
 	struct drm_fbdev_cma		*fbdev;
-
-	struct drm_gem_cma_object	*gem;
-	bool				enabled;
-	bool				suspended;
 };
 
 int mxsfb_setup_crtc(struct drm_device *dev);
-- 
1.7.9.5

